torrt/bots/telegram_bot.py

```diff
@@ -99,28 +117,53 @@ def handle_ask_download_path(self, bot, update, user_data):
             if path == '.':
                 path = None
 
-            torrents_count = len(config.load()['torrents'])
-            add_torrent_from_url(torrent_url, download_to=path)
-            if len(config.load()['torrents']) > torrents_count:
-                update.message.reply_text('Torrent from `%s` was added' % torrent_url)
+            torrents_count = len(get_registered_torrents())
+            try:
+                add_torrent_from_url(torrent_url, download_to=path)
+            except Exception as e:
+                logging.error('Torrent was not added: {}'.format(e))
+                update.message.reply_text('Error was occurred during registering torrent.',
+                                          reply_markup=ReplyKeyboardRemove())
+            if len(get_registered_torrents()) > torrents_count:
+                update.message.reply_text('Torrent from `%s` was added' % torrent_url,
+                                          reply_markup=ReplyKeyboardRemove())
             else:
-                update.message.reply_text('Torrent was not added.')
+                update.message.reply_text('Torrent was not added.', reply_markup=ReplyKeyboardRemove())
         return ConversationHandler.END
 
     def cancel_handler(self, bot, update):
-        update.message.reply_text('Bye! I hope to see tou again.',
+        update.message.reply_text('Bye! I hope to see you again.',
                                   reply_markup=ReplyKeyboardRemove())
 
         return ConversationHandler.END
 
+    def handle_remove_torrents(self, bot, update):
+        data = update.callback_query.data
+        splitted_data = data.split(':')[1:]
+        torrent_hash = splitted_data.pop(0)
+        if not splitted_data:
+            keyboard = InlineKeyboardMarkup([[InlineKeyboardButton(text="Yes", callback_data=data + ':1'),
+                                              InlineKeyboardButton(text="No", callback_data=data + ':0')]])
+            update.callback_query.message.reply_text('Do you want to delete data?', reply_markup=keyboard)
+        else:
+            torrent_data = get_registered_torrents().get(torrent_hash)
+            if torrent_data:
+                remove_torrent(torrent_hash, with_data=bool(int(splitted_data[0])))
+                update.callback_query.message.reply_text('Torrent `{}` was removed'.format(torrent_data['name']))
+            else:
+                update.callback_query.message.reply_text('Torrent not found. Try one more time with /remove')
```

Comment:
It could happen only if user push button with already removed torrent from any previous message. If he start command /remove one more time he did not see this torrent in a list.